colcon_package_information/verb/info.py

```diff
@@ -20,28 +22,36 @@ def __init__(self):  # noqa: D107
 
     def add_arguments(self, *, parser):  # noqa: D102
         parser.add_argument(
-            'path', nargs='?',
-            help='Specific path to check for package (ignore other discovery '
-                 'arguments)')
+            'package_names', nargs='*', metavar='PKG_NAME',
+            help='Only show the information of a subset of packages')
         add_packages_arguments(parser)
 
     def main(self, *, context):  # noqa: D102
-        # modify args to match path discovery extension
-        if context.args.path is not None:
-            context.args.paths = [context.args.path]
-
         descriptors = get_package_descriptors(
             context.args, additional_argument_names=['*'])
         decorators = get_decorators(descriptors)
-        if context.args.path is None:
-            add_recursive_dependencies(
-                decorators, recursive_categories=('run', ))
-            select_package_decorators(context.args, decorators)
+        add_recursive_dependencies(
+            decorators, recursive_categories=('run', ))
+        select_package_decorators(context.args, decorators)
 
-        if not descriptors:
-            return 'No package found'
+        if context.args.package_names:
+            package_names = {d.descriptor.name for d in decorators}
```

Comment:
nit: I think it would be easier to understand this later if this was named `all_package_names` instead.

Reply:
Sounds good to me. Renamed: https://github.com/colcon/colcon-package-information/pull/5/files#diff-570822ec77e6e8526e110c1c2e38185dR38